resources/images/filters.go

```diff
@@ -110,6 +113,65 @@ func (*Filters) Text(text string, options ...any) gift.Filter {
 	}
 }
 
+// Padding creates a filter that resizes the image canvas without resizing the
+// image. The last argument is the canvas color, expressed as an RGB or RGBA
+// hexadecimal color. The default value is `ffffffff` (opaque white). The
+// preceding arguments are the padding values, in pixels, using the CSS
+// shorthand property syntax. Negative padding values will crop the image. The
+// signature is images.Padding V1 [V2] [V3] [V4] [COLOR].
+func (*Filters) Padding(args ...any) gift.Filter {
```

Comment:
I like the overall idea of the Padding func. But I think the variadic function setup becomes too complicated with 5 different arguments.

I suggest

* using a dict with top, right, bottom, left, color
* create a struct with these and use mapstructure.WeakDecode
* But make sure to pass the original dict (the one provided from the user) to newFilterOpts (this allows us to add fields to the options struct without breaking the hash)

Reply:
This is a polite request to reconsider. In my view the most common cases are:

```go-html-template
{{ $filter := images.Padding 20 }}
{{ $filter := images.Padding 20 40 }}
```

Also, the CSS padding shorthand should be familiar to most site and theme authors, and I can certainly add links to mozilla docs describing the notation.

I'm fine with the options map; it just seems like a little more work compared to some of the other filters, excluding text and overlay.

Reply:
Can you guarantee that there will be no more options coming in the future here?

Reply:
What if you could have both ... So we would check for the common cases (1 or 2 integers) as the special case.

Reply:
Yea, ok, we could introduce the dict if it gets more complicated.